superset/assets/src/visualizations/WorldMap/transformProps.js

```diff
@@ -0,0 +1,11 @@
+export default function transformProps(basicChartInput) {
```

Comment:
that's an interesting idea. I think that doing that in another PR could be okay tho, it would probably be a decent amount of work to enumerate all of the options there. it's basically defining the formData interface / would be a precursor to the `type script` interface.

Reply:
It could be part of the `BasicChartInput`. 

For example, `class PieChartInput extends BasicChartInput` then in the constructor you parse `formData` with `new PieChartFormData(formData)`.